xdem/volume.py

```diff
@@ -266,3 +342,146 @@ def hypsometric_interpolation(voided_ddem: Union[np.ndarray, np.ma.masked_array]
     assert output is not None
 
     return output
+
+
+def local_hypsometric_interpolation(voided_ddem: Union[np.ndarray, np.ma.masked_array],
```

Comment:
In terms of performance/structure, I suggest moving away from the method of using a `mask` with multiple values for each feature of a geometry. It's almost instantaneous to generate a mask for a single feature, so the loop over shapefile features should be outside this function (especially given that this is the `local_hypsometric_method`).
In particular, it will be more easy/efficient to subdivide tasks and parallelize using multiprocessing this way (echoes Erik's comment below),

Reply:
Interesting. @adehecq  and I actually argued (without empiric basis) for the opposite, that we wanted to avoid multiple rasterize calls to improve performance! Should we benchmark the two approaches, or do you already know that your suggested approach (and the one I am already using for the `dDEM.interpolate("local_hypsometric")`) is more efficient, @rhugonnet, or should we try to figure that out?

I guess the most important part is that it is understandable for the user. 

What if, instead of giving a `mask` with integers representing the glaciers, the function just takes an iterable of `shapely.geometry.base.BaseGeometry` instances and will fill every outline that meets the `min_coverage` criterion?

Reply:
It would be worth testing both approaches indeed.
But I think the approach I implemented works better for a large raster with small geometries. In that case, I disagree with you @rhugonnet that generating a mask is almost instantaneous, as one need to create e.g. a 20k x 20k array each time for only a few valid pixels...
Ideally, for each geometry, one would crop the raster, create the associated mask and pass it to the generic hypsometric approach.

Reply:
I've added an issue to remember to test both approaches.